kube-runtime/src/scheduler.rs

```diff
@@ -126,6 +126,53 @@ impl<'a, T: Hash + Eq + Clone, R> SchedulerProj<'a, T, R> {
             }
         }
     }
+
+    /// Attempt to retrieve a message from queue and mark it as pending.
+    pub fn pop_queue_message_into_pending(&mut self, cx: &mut Context<'_>) -> Poll<T> {
+        loop {
+            match self.queue.poll_expired(cx) {
+                Poll::Ready(Some(msg)) => {
+                    let msg = msg.into_inner();
+                    self.pending.insert(msg);
+                }
+                Poll::Ready(None) | Poll::Pending => break Poll::Pending,
+            }
+        }
+    }
+}
+
+/// See [`Scheduler::hold`]
+pub struct Hold<'a, T, R> {
+    scheduler: Pin<&'a mut Scheduler<T, R>>,
+}
```

Comment:
If this reference to the scheduler is being held across potentially much longer time periods now (particularly if users are being overaggressive with a low concurrency value), do we have anything that ensures there is backpressure on the applier?

(there is a rather complicated test for this on the applier side in https://github.com/kube-rs/kube/blob/b6e13e4c4708d14831002fc6e46194dc6abe6ac3/kube-runtime/src/controller/mod.rs#L1343-L1348 and i'm not sure it's sufficient to catch this new buffer point, but maybe it is?)